### PR TITLE
refactor: move to electron-vite for build, clean up CI

### PR DESCRIPTION
**Pull request by Benjozork**
## Summary of Changes

- Entirely replaces webpack with electron-vite
- Cleans up PR CI

### CODE REVIEW

**Comment by brewers10** [comment]
Quality Assurance Tester Report

Discord: brewers_10
Object of testing: #462 
Tier of Testing: 1
Date: 14/04/2024

Testing Process: 
Installed installer
Installed stable versions of A32NX, B74S, KFBW and Simbridge. Simbridge loads perfectly fine and aircraft/scenery work with sim.

Negatives:       
N/A

Testing Results: 
Passed

Conclusions: 
LGTM

Media:
N/A

File: src/renderer/index.tsx [hunk 1]
```diff
@@ -48,6 +48,8 @@ import 'simplebar-react/dist/simplebar.min.css';
 import './index.scss';
 import { Button, ButtonType } from 'renderer/components/Button';
 
+console.log('a');
```
Comment (alepouna) [inline src/renderer/index.tsx:51]:
```suggestion

```
Debug log?

**Review by alepouna — CHANGES_REQUESTED** [review]

**Comment by alepouna** [comment]
Quality Assurance Tester Report

Discord: alepouna
Object of testing: #462
Date: 23/04/2024

Testing Process:
- Installed installer

Negatives:
Descriptions are missing from addons, 'Error' when hovering on already installed items.
image

Testing Results:
Not Passed

Console logs:
Spam of Sentry errors such as
> index-CbxzvBJf.js:7930 Fetch API cannot load sentry-ipc://sentry-electron.scope/sentry_key. URL scheme "sentry-ipc" is not supported.
> index-CbxzvBJf.js:7649 Sentry SDK failed to establish connection with the Electron main process.
  - Ensure you have initialized the SDK in the main process
  - If your renderers use custom sessions, be sure to set 'getSessions' in the main process options
  - If you are bundling your main process code and using Electron < v5, you'll need to manually configure a preload script
 
No other error logs.

**Comment by noonebling** [comment]
Quality Assurance Trainee Report

Discord          : noonebling
Object of testing: #462 
Tier of Testing  : 1
Date             : 24/04/2024

Testing Process: 
Installed Installer.
Installed different versions of addons in the installer

Negatives:
Descriptions are missing in all addons
image

The slider bars in Settings page do not change appearance when kept to the right (On position)
image


Testing Results: 
Not Passed

**Comment by alepouna** [comment]
Quality Assurance Tester Report

Discord: alepouna
Object of testing: #462
Date: 27/04/2024

Testing Process:
- Installed installer 
- Test various addons and updating
- Test custom configs 


Testing Results:
Passed

**Comment by Popespice** [comment]
Quality Assurance Tester Report

Discord          : Popespice
Object of testing: #462 
Tier of Testing  : 1
Date             : 28/4/2024

Testing Process: 
1. Deleted Appdata existing installer and supporting folder
2. Installed new installer from PR build
3. Opened Installer
4. Navigate to Settings
5. Click "third party licenses"
6. Observe popup that states "requested file not found"

Negatives:       
(Any issues, doubts. - type N/A if none are found)

Testing Results: 
Not Passed

Conclusions: 
Not working as expected

Media:       
Quality Assurance Tester/Trainee Report

Discord          : (name#1234)
Object of testing: (#PR Number)
Tier of Testing  : (1/2)
Date             : (DD/MM/YYYY)

Testing Process: 
(Detail is needed)

Negatives:       
(Any issues, doubts. - type N/A if none are found)

Testing Results: 
(Passed/Not Passed)

Conclusions: 
(Notes made, the positives in the PR, and anything extra you would like 
to mention - Delete Line if Nothing Extra is to be mentioned)

Media:
N/A

**Review by alepouna — APPROVED** [review]

File: src/renderer/utils/Directories.ts [hunk 2]
```diff
@@ -60,11 +69,11 @@ export class Directories {
   }
 
   static inPackagesMicrosoftStore(targetDir: string): string {
-    return path.join(process.env.LOCALAPPDATA, MSFS_APPDATA_PATH, this.sanitize(targetDir));
+    return path.join(Directories.localAppData(), MSFS_APPDATA_PATH, this.sanitize(targetDir));
   }
 
   static inPackagesSteam(targetDir: string): string {
-    return path.join(process.env.APPDATA, MSFS_STEAM_PATH, this.sanitize(targetDir));
+    return path.join(Directories.localAppData(), MSFS_STEAM_PATH, this.sanitize(targetDir));
```
Comment (FoxtrotSierra6829) [inline src/renderer/utils/Directories.ts:76]: Steam installs into in AppData/Roaming/

Comment (Benjozork) [reply src/renderer/utils/Directories.ts:76]: Oops

**Comment by alepouna** [comment]
Quality Assurance Tester Report

Discord: alepouna
Object of testing: #462
Date: 22/08/2024

Testing Process:
- Installed installer 
- Test install various addons (A32NX, big bertha, FSLTL, Salty) and updating. Test CDN and direct.
- Test custom configs 
- Check 3rd party licenses page loads fine

Testing Results:
Passed

File: src/renderer/utils/Directories.ts (hunk 2 above)
Comment (Benjozork) [reply src/renderer/utils/Directories.ts]: Fixed

**Pull request merged**